### PR TITLE
fix bug embedding_size mismatch with hidden_size in electra model test

### PR DESCRIPTION
**Pull request by kaixuanliu**
@ArthurZucker @Cyrilvallez pls help review. Thx!

### CODE REVIEW

**Comment by kaixuanliu** [comment]
When I run unit test: `tests/models/electra/test_modeling_electra.py::ElectraModelTest::test_sdpa_padding_matches_padding_free_with_position_ids`, it gets flakey result: fails sometimes and sometimes it can pass. After investigation, I found that in this case, `hidden_size=32`(explicitly set: [L55](https://github.com/huggingface/transformers/blob/main/tests/models/electra/test_modeling_electra.py#L55) ) while `embedding_size=128`(set by default). In this model, `ElectraGeneratorPredictions` has a dense(hidden_size → embedding_size) layer. With embedding_size=128, it expands the 32-dim SDPA small numerical differences to 128 dims, then lm_head(128→99) accumulates BF16 rounding over 128 dot products, hence it will exceed the tolerance sometimes(depending on the random encoder_attention_mask: [L156](https://github.com/huggingface/transformers/blob/main/tests/models/electra/test_modeling_electra.py#L156) ). W/ this PR, it can get passed every time. And BTW it seems more reasonable to use a embedding_size value <= hidden_size

**Comment by kaixuanliu** [comment]
@ydshieh Hi, can you help review the test case related PR? Thx!

**Review by ydshieh — APPROVED** [review]
Thank you!
